libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -607,6 +607,26 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 
 		break;
 	}
+	case FunctionType::Kind::KECCAK256:
+	{
+		solAssert(arguments.size() == 1, "");
+
+		ArrayType const* arrayType = TypeProvider::bytesMemory();
+		string const& array = m_context.newYulVariable();
```

Comment:
Is there a benefit of using `const&` here?

Reply:
We never want nor should modify it after creation, the `const` assures this. There might also be optimization advantages.

Reply:
ok, but what about the `&`?

Reply:
I suppose it's not doing a copy of the data either way.. Do you see a disadvantage to it?

Reply:
The disadvantage is that you have to think about whether `newYulVariable` might return via reference and the lifetime of that reference in that case.